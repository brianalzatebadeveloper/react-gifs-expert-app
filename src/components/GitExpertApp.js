import React, { useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['the rock']);

  return (
    <>
     <h2>GifExpertApp</h2>
     <AddCategory setCategories={ setCategories } />
     <hr />
     <ol>
       { 
          categories.map( (category, i) => (
             <GifGrid 
                key={ category }
                category={ category } 
             />
          )) 
       }
     </ol>
    </>
  );

}

export default GifExpertApp;