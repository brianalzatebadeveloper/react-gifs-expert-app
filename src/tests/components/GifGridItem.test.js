import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from  '../../components/GifGridItem';


describe('Pruebas del componente GifGridItem.js', () => {

  const image = {
    id: '1', 
    title: 'Can You Smell The Rock GIF by WWE', 
    url: 'https://media1.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19exqupm9xvcfjarglgzwksmozb9542poktfd6af8i&rid=giphy.gif&ct=g'
  };
  const wrapper = shallow( <GifGridItem { ...image } /> );

  test('Debe mostrar el componente <GifGridItem /> correctamente', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe tener un parrafo con el title', () => {

    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( image.title );

  });

  test('Debe tener la imagen igual al url y alt de los props', () => {

    const img = wrapper.find('img');
    expect( img.props().alt ).toBe( image.title );
    expect( img.prop('src') ).toBe( image.url );

  });

  test('Debe tener animate_fadeIn', () => {

    const div = wrapper.find('div');
    const className = div.props().className;
    expect( className.includes('animate__fadeIn') ).toBe( true );

  });

});