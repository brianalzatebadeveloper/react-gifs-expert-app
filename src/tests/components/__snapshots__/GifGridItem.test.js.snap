// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem.js Debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Can You Smell The Rock GIF by WWE"
    src="https://media1.giphy.com/media/l0HUg6Ypas42ubkXu/giphy.gif?cid=0858df19exqupm9xvcfjarglgzwksmozb9542poktfd6af8i&rid=giphy.gif&ct=g"
  />
  <p>
    Can You Smell The Rock GIF by WWE
  </p>
</div>
`;
